cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGenerator.scala

```diff
@@ -0,0 +1,606 @@
+          queryBuilder.append(s"\n) $groupedLookupSourceVar\n")
+          val resultAfterLookup = parseLookup(group.lookups, newExprMap, queryBuilder, true)
+          ParsedResult(Seq.empty, resultAfterLookup.exprMap)
+        } else {
+          ParsedResult(Seq.empty, producedExprs.result().toMap)
+        }
+
+      case None => ParsedResult(Seq(""), exprMap)
+    }
+  }
+
+  private def parseSelect(selectOpt: Option[SelectStatement],
+                          exprMap: Map[String, FieldExpr], query: Query,
+                          queryBuilder: StringBuilder): ParsedResult = {
+    selectOpt match {
+      case Some(select) =>
+        val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
+
+        val orderStrs = select.orderOn.zip(select.order).map {
+          case (orderOn, order) =>
+            val expr = exprMap(orderOn.name).defExpr
+            val orderStr = if (order == SortOrder.DSC) "desc" else ""
+            s"${expr} $orderStr"
+        }
+
+        val orderStr =
+          if (orderStrs.nonEmpty) {
+            orderStrs.mkString("order by ", ",", "")
+          } else {
+            s""
+          }
+
+        val limitStr = s"limit ${select.limit}"
+        appendIfNotEmpty(queryBuilder, orderStr)
+        if (select.limit != 0) {
+          appendIfNotEmpty(queryBuilder, limitStr)
+        }
+
+        if (select.fields.isEmpty || query.hasUnnest || query.hasGroup) {
+          producedExprs ++= exprMap
+        } else {
+          select.fields.foreach {
+            case AllField =>
+              producedExprs ++= exprMap
+            case field =>
+              producedExprs += field.name -> exprMap(field.name)
+          }
+        }
+
+        val newExprMap = producedExprs.result().toMap
+        val projectStr = if (select.fields.isEmpty) {
+          if (query.hasUnnest || query.hasGroup) {
+            parseProject(exprMap)
+          } else {
+            s"select *"
+          }
+        } else {
+          parseProject(exprMap)
+        }
+        queryBuilder.insert(0, projectStr + "\n")
+        ParsedResult(Seq.empty, newExprMap)
+
+      case None =>
+        val projectStr =
+          if (query.hasUnnest || query.hasGroup) {
+            parseProject(exprMap)
+          } else {
+            s"select *"
+          }
+        queryBuilder.insert(0, projectStr + "\n")
+        ParsedResult(Seq.empty, exprMap)
+    }
+  }
+
+  private def parseProject(exprMap: Map[String, FieldExpr]): String = {
+    exprMap.filter {
+      case (field, expr) => field != "*" && expr.refExpr != sourceVar
+    }.map {
+      case (field, expr) =>
+        if (s"${expr.defExpr}" == s"$quote$second$quote" || s"${expr.defExpr}" == s"$quote$minute$quote" || s"${expr.defExpr}" == s"$quote$hour$quote" || s"${expr.defExpr}" == s"$quote$day$quote" || s"${expr.defExpr}" == s"$quote$month$quote" || s"${expr.defExpr}" == s"$quote$year$quote"){
```

Comment:
why is this part so complicated? and I even see the `create_at` appears. it should be general purpose generator.